public/app/plugins/datasource/input/module.ts

```diff
@@ -6,6 +6,6 @@ import { InputQueryEditor } from './InputQueryEditor';
 import { InputConfigEditor } from './InputConfigEditor';
 import { InputOptions, InputQuery } from './types';
 
-export const plugin = new DataSourcePlugin<InputQuery, InputOptions>(InputDatasource)
+export const plugin = new DataSourcePlugin<InputDatasource, InputQuery, InputOptions>(InputDatasource)
```

Comment:
It feels like we should be able to simplify with some flavor of generics capture... but I could not find a solution.  The fact that we are passing in a type as the generic and the instance is a little weird.